src/build_log.h

```diff
@@ -76,8 +86,14 @@ struct BuildLog {
              int start_time, int end_time, TimeStamp mtime);
   };
 
+#if __cplusplus >= 201103L || (defined(_MSC_VER) && _MSC_VER >= 1900)
+  typedef std::shared_ptr<LogEntry> LogEntryPtr;
+#else
+  typedef LogEntry* LogEntryPtr;
+#endif
+
   /// Lookup a previously-run command by its output path.
-  LogEntry* LookupByOutput(const std::string& path);
+  LogEntryPtr LookupByOutput(const std::string& path);
```

Comment:
return by const& to avoid incrementing the refcount.

Reply:
I could not return the pointer as `const&` because the rvalue is returned.

https://github.com/ninja-build/ninja/blob/1d4034f0ac2b5cfb809b5ab983d47c3cb2c78415/src/build_log.cc#L377-L382